src/ucs/memory/rcache.c

```diff
@@ -288,6 +289,52 @@ static void ucs_rcache_find_regions(ucs_rcache_t *rcache, ucs_pgt_addr_t from,
                              ucs_rcache_region_collect_callback, list);
 }
 
+/* LRU spinlock must be held */
+static inline void
+ucs_rcache_region_lru_add(ucs_rcache_t *rcache, ucs_rcache_region_t *region)
+{
+    if (region->flags & UCS_RCACHE_REGION_FLAG_LRU) {
+        return;
+    }
+
+    ucs_rcache_region_trace(rcache, region, "lru add");
+    ucs_list_add_tail(&rcache->lru.list, &region->lru_list);
+    ++rcache->lru.count;
+    region->flags |= UCS_RCACHE_REGION_FLAG_LRU;
```

Comment:
if we modify this flag only under lru lock, I think it can race with other function modifying region->flags